### PR TITLE
[SDK][Doc] Update dialect documentation page to describe join support

### PR DESCRIPTION
**Pull request by saugatt**
join support

### CODE REVIEW

File: docs/dialect.md [hunk 1]
```diff
@@ -133,3 +133,40 @@ The <span style="font-family: courier new">argument</span> element is an opt
...
+
+  - __NoNullCheck__
+      Does not check that the value of LHS and RHS is null. Formula: `(lhs [!]= rhs)`
+  - __Keyword__
+      Uses `DISTINCT` keyword for comparision. Formula: `(lhs IS [NOT ]DISTINCT FROM rhs)`.
+  - __Operator__
+    Uses `<=>` for comparions. Formula:`([NOT] (lhs <=> rhs))`
+
+  By default, format is distinct checks that the value for LHS and RHS is not null.   <br/>
+  Formula :`((lhs [!]= rhs) OR[AND] (lhs IS [NOT] NULL AND[OR] rhs IS [NOT] NULL))`
```
Comment (jsun28) [inline docs/dialect.md:149]: I think the Formula should be also listed as a possible value for this property, after NoNullCheck, Keyword, and Operator.

Comment (pvanderknyff) [reply docs/dialect.md:149]: +1

File: docs/dialect.md [hunk 2]
```diff
@@ -133,3 +133,40 @@ The <span style="font-family: courier new">argument</span> element is an opt
...
+  
+  ```xml 
+  <format-is-distinct value='NoNullCheck' />
+  ```
+ 
+  **Supported Joins** <br/>
```
Comment (rosswbrown) [inline docs/dialect.md:155]: I would put this section before Format is distinct, its more important in general

File: docs/dialect.md [hunk 3]
```diff
@@ -133,3 +133,40 @@ The <span style="font-family: courier new">argument</span> element is an opt
...
+          <part name='Left' />
+          <part name='Right' />
+          <part name='Full' />
+          <part name='Cross' />
+        </supported-joins>
+      </sql-format>
+      </dialect>
+    ...
+```   
+  **CAP QUERY JOIN**  <br/>
```
Comment (rosswbrown) [inline docs/dialect.md:171]:
```suggestion
  **Join Capabilities**  <br/>
```

File: docs/dialect.md [hunk 4]
```diff
@@ -133,3 +133,40 @@ The <span style="font-family: courier new">argument</span> element is an opt
...
+  You can set the capabilities for joins in your manifest file. See capabilities for the Query section to see join capabilities  [here]({{ site.baseurl }}/docs/capabilities).
```
Comment (rosswbrown) [inline docs/dialect.md:172]: Join usage is also defined by capabilities in the manifest file. See the Query section for join relation capabilities here.

File: docs/dialect.md (hunk 1 above)
Comment (saugatt) [reply docs/dialect.md:149]: Good catch made the changes.

File: docs/dialect.md (hunk 2 above)
Comment (saugatt) [reply docs/dialect.md:155]: Makes sense, made the changes.

File: docs/dialect.md (hunk 4 above)
Comment (saugatt) [reply docs/dialect.md]: This seems clear.

**Pull request merged**